File: Top Interview 150/Two Pointer/11-Container With Most Water.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int l = 0;
        int r = height.size() - 1;
        int maxwater = 0;
        while (l < r)
        {
            int area = min(height[l], height[r]) * (r - l);
            maxwater = max(maxwater, area);
            if (height[l] <= height[r])
                l++;
            else
                r--;
        }
        return maxwater;
    }
};

int main()
{

    return 0;
}

File: Strivers A2Z DSA Sheet/11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 8/7/2026, 7:17:37 PM
class Solution {
    public int maxArea(int[] height) {
        int l=0;
        int r = height.length - 1;
        int maxwater = 0;

        while(l<r){
            int area = Math.min(height[l], height[r]) * (r-l);
            maxwater = Math.max(maxwater, area);
            if(height[l]<=height[r]) l++;
            else r--;
        }
        return maxwater;
    }
}